src/kartograf/atom_mapper.py

```diff
@@ -855,6 +854,38 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    def _split_protein_components_molecules(self, protein: ProteinComponent) -> list[ProteinComponent]:
+        """
+        Aims at splitting a protein component into different protein components based on the
+        connectivity of the molecules that compose it. Useful for mapping multimer components
+        or proteins with structural waters or similarly.
+
+        This returns a ``ProteinComponent`` object with the name attribute indicating the starting
+        index in the original component.
+        """
+        from rdkit.Chem.rdmolops import GetMolFrags
+        rdmol = protein.to_rdkit()
+        fragments_indices = GetMolFrags(rdmol)
+        components = []
+        for index_tuple in fragments_indices:
+            edit_rdmol_frag = Chem.EditableMol(rdmol)
+            remove_indices = []
+            for atom in rdmol.GetAtoms():
+                atom_index = atom.GetIdx()
+                if not (atom_index in index_tuple):
+                    remove_indices.append(atom_index)
+            # Need to remove separately https://github.com/rdkit/rdkit/issues/1366
+            for atom_idx in sorted(remove_indices, reverse=True):
+                edit_rdmol_frag.RemoveAtom(atom_idx)
+            #  Create component with the remaining molecule
+            frag_rdmol = edit_rdmol_frag.GetMol()
+            Chem.SanitizeMol(frag_rdmol)
+            # FIXME: We are storing the starting index in name, this is super hacky :/
+            protein_comp = ProteinComponent(frag_rdmol, name=f"frag_{index_tuple[0]}")
+            components.append(protein_comp)
```

Comment:
```suggestion
        index_tuples = []
        fragments = GetMolFrags(rdmol, asMols=True, sanitizeFrags=True, fragsMolAtomMapping=index_tuples)
        components = [ProteinComponent(fragment, name=f"frag_{index_tuple[0]}") for fragment, index_tuple in zip(fragments, index_tuples)]
```

Reply:
This should be much faster and give the same result.

Reply:
Ah this is cool. I did read the options for the `GetMolFrags` object, but I couldn't really tell if they were going to achieve the same. The tests passing should be enough to check this, but I guess we could also double check if we are testing enough. Thanks!